### PR TITLE
Fixing for Rml 5.* releases Vulkan issues

### PR DESCRIPTION
**Pull request by wh1t3lord**
Finally, I got two free hours and made this! 

So, I added new file in vulkan's folder for debugging because modern Vulkan API requires to use version api higher or equal to 1.1 otherwise we will get in log validation error, but still we keep original vulkan.h that was generated for vulkan 1.0 for supporting maximum 'devices' (and probably users).

Fix was simple, as I said, so we had render pass and function that clearing depth stencil it calls **outside** but by our design it supposed to be **inside** obviously I used inside version of that function and now all works fine.

And minor thing is I changed initialization flags for depth stencil because it forces to use only two variants of flags so yeah Vulkan API changes quite fast...

Thanks again for reporting by @skaarj1989 and @DrJedikiah 

https://github.com/mikke89/RmlUi/issues/581
https://github.com/mikke89/RmlUi/issues/558

### CODE REVIEW

**Comment by wh1t3lord** [comment]
About using new version of vulkan.h is simple. In folder where vulkan.h is located there is one file called 'vulkan_modern.h' just rename it to vulkan.h (but before that rename the original vulkan.h file) and than just compile your solution.

And I added preprocessor to header file that is called RMLUI_VK_API_VERSION just set from VK_API_VERSION_1_0 to VK_API_VERSION_1_1. After debugging restore the things to original state.

File: Backends/RmlUi_Renderer_VK.cpp [hunk 1]
```diff
@@ -785,8 +787,18 @@ void RenderInterface_VK::SetViewport(int width, int height)
 	if (window_extent.width == 0 || window_extent.height == 0)
 		return;
 
+#ifdef RMLUI_DEBUG
```
Comment (mikke89) [inline Backends/RmlUi_Renderer_VK.cpp:790]: I suggest that we use `RMLUI_VK_DEBUG` like the rest of the debug logging here.

**Review by mikke89 — COMMENTED** [review]
Thank you for the fixups! This seems to be working well for me.

About the `vulkan_modern.h` header: I don't think we should add this unused header. Its purpose will be unclear to others, and it will most likely sit around and become stale.

If it helps with debugging, I don't mind bumping the Vulkan minimum to 1.1. Looks like it is very widely supported for GPUs that support Vulkan. Otherwise, we can simply remove it for now.

**Comment by wh1t3lord** [comment]
@mikke89 my corrections I pushed

**Comment by mikke89** [comment]
Thanks a lot for the pull request! Looks good to me, and fixes the validation errors on my end. Hopefully it fixes the glitches that others experienced too.

**Pull request merged**